network/network.go

```diff
@@ -411,10 +412,28 @@ func (n *Network) validateNodeDID(nodeDID did.DID) error {
 	// Check if the DID document has a resolvable NutsComm endpoint
 	serviceResolver := doc.NewServiceResolver(n.didDocumentResolver)
 	serviceRef := doc.MakeServiceReference(nodeDID, transport.NutsCommServiceType)
-	_, err = serviceResolver.Resolve(serviceRef, doc.DefaultMaxServiceReferenceDepth)
+	nutsCommService, err := serviceResolver.Resolve(serviceRef, doc.DefaultMaxServiceReferenceDepth)
 	if err != nil {
 		return fmt.Errorf("unable to resolve %s service endpoint, register it on the DID document (did=%s): %v", transport.NutsCommServiceType, nodeDID, err)
 	}
+
+	// Self-authentication requires a certificate. Running non-TLS is evaluated in Configure
+	if n.certificate.Leaf == nil { // n.config.DisableNodeAuthentication is for incoming connections.
+		return nil
+	}
+
+	// Check if the NutsComm endpoint matches a SAN on the certificate
+	var nutsCommURLStr string
+	if err = nutsCommService.UnmarshalServiceEndpoint(&nutsCommURLStr); err != nil {
+		return fmt.Errorf("invalid %s service endpoint: %w", transport.NutsCommServiceType, err)
+	}
+	nutsCommURL, err := url.Parse(nutsCommURLStr)
```

Comment:
You can use `transport.ParseNutsCommAddress(nutsCommStr)`

Reply:
Debugging setups probably don't use strict mode?

Now you're also missing the validation of `grpc://` (which `ParseNutsCommAddress` does)

Reply:
Reserved addresses and IPs are not prohibited in the RFCs. Shouldn't they be added there first? (I'm assuming you can't register these addresses on a PKIo certificate anyway?)

Reply:
Our testing makes use of reserved addresses (localhost, local, etc.). I've fixed everything for this repo, but e2e still needs to be done